doc/adrs/157_improve_react_component_representations_rendering.adoc

```diff
@@ -0,0 +1,175 @@
+= [ADR-157] Improve React component representations rendering
+
+
+== Context
+
+
+When we want to access a representation we have to subscribe to it.
+
+
+Each time a representation changes, a new payload is sent to the subscribers of this representation.
+
+
+This guarantee that all subscribers of a representation are in synch with the actual data of the representation
+
+
+== The issue
+
+
+Our front-end is based on React, we have a main React Component for each representation and then a smaller React Component for parts of the representation.
+
+
+For a diagram based Representation we have the main component ```DiagramRenderer``` and as children we have ```nodes``` and ```edges```.
+
+
+With the actual implementation, when we receive a new payload to update the representation we also update all the components under it.
+
+
+This causes a noticeable impact on performance with representation rendering lots of elements and particularly if they use lots of interactions.
```

Comment:
Not sure what "use lots of interactions" means?

Reply:
Yes I'll change the phrasing, I wanted to point out representation that have more way to interact with than others (like diagrams)